File: pinyougou_sellergoods_interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbBrand;
import entity.PageResult;

import java.util.List;

/**
 * @ClassName: BrandService
 * @Description: 查询品牌列表
 * @Author: XuZhao
 * @CreateDate: 19/03/09$ 下午 07:41$
 */
public interface BrandService {
    //查询品牌列表
    public List<TbBrand> findAll();

    //分页查询品牌列表
    PageResult findPage(Integer pageNum, Integer pageSize);

    //新增品牌
    void add(TbBrand tbBrand);

    //根据id查询
    TbBrand findOne(Long id);

    //修改品牌
    void update(TbBrand tbBrand);

    //批量删除
    void delete(Long[] ids);

    //品牌分页查询
    PageResult search(TbBrand brand, Integer pageNum, Integer pageSize);
}

File: pinyougou_manager_web/src/main/webapp/js/service/brandService.js
//定义服务层
app.service("brandService",function ($http) {
    //查询所有品牌
    this.findAll=function () {
        return $http.get('../brand/findAll.do');
    }
    //分页查询
    this.findPage=function (pageNum,pageSize) {
        return $http.get("../brand/findPage.do?pageNum="+pageNum+"&pageSize="+pageSize);
    }
    //条件分页查询
    this.search=function (pageNum,pageSize,searchEntity) {
        return $http.post("../brand/search.do?pageNum="+pageNum+"&pageSize="+pageSize,searchEntity);
    }

    //根据id回显数据
    this.findOne=function (id) {
        return $http.get("../brand/findOne.do?id="+id);
    }
    //新增品牌
    this.add=function (entity) {
        return $http.post("../brand/add.do",entity);
    }
    //修改品牌
    this.update=function (entity) {
        return $http.post("../brand/update.do",entity);
    }
    //批量删除
    this.dele=function (ids) {
        return $http.get("../brand/delete.do?ids="+ids);
    }
    //查询模板关联的品牌列表
    this.selectBrandOptions=function () {
        return $http.get("../brand/selectBrandOptions.do");
    }

})